.github/workflows/automated-prs-manager.yaml

```diff
@@ -0,0 +1,95 @@
+name: Automated PRs Manager
+
+on:
+  schedule:
+  - cron: "0 */6 * * *" # every 6 hours
+  workflow_dispatch: {}
+
+jobs:
+  list-prs:
+    runs-on: ubuntu-latest
+    outputs:
+      prs: ${{ steps.list-prs.outputs.prs }}
+    env:
+      GH_TOKEN: ${{ secrets.REPLICATED_GH_PAT }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: List PRs
+        id: list-prs
+        run: |
+          set -euo pipefail
+
+          # list prs that are less than 24h old and exclude prs from forks
+
+          prs=$(
+            gh pr list \
+              --author 'dependabot[bot]' \
+              --json url,createdAt,headRefName,headRepository,headRepositoryOwner \
+              -q '.[] | select((.createdAt | fromdateiso8601 > now - 24*60*60) and .headRepositoryOwner.login == "replicatedhq" and .headRepository.name == "troubleshoot")'
+          )
+
+          echo "prs=$prs" >> "$GITHUB_OUTPUT"
+
+  process-prs:
+    needs: list-prs
+    runs-on: ubuntu-latest
+    if: needs.list-prs.outputs.prs != '[]'
+    strategy:
+      matrix:
+        pr: ${{ fromJson(needs.list-prs.outputs.prs) }}
+      fail-fast: false
+      max-parallel: 1
+    env:
+      GH_TOKEN: ${{ secrets.REPLICATED_GH_PAT }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ matrix.pr.headRefName }}
+
+      - name: Process PR
+        run: |
+          set -euo pipefail
+
+          echo "Ensuring required labels..."
+          gh pr edit "${{ matrix.pr.url }}" --add-label "type::security"
+
+          echo "Checking status of tests..."
+          run_id=$(gh run list --branch "${{ matrix.pr.headRefName }}" --workflow build-test --limit 1 --json databaseId -q '.[0].databaseId')
+
+          # If there are still pending jobs, skip.
+
+          num_of_pending_jobs=$(gh run view "$run_id" --json jobs -q '.jobs[] | select(.conclusion == "") | .name' | wc -l)
+          if [ "$num_of_pending_jobs" -gt 0 ]; then
+            echo "There are still pending jobs. Skipping."
+            exit 0
+          fi
+
+          # If all checks passed, approve and merge.
+
+          if gh pr checks "${{ matrix.pr.url }}" ; then
```

Comment:
i believe i encountered a race condition when trying to only validate the pr checks this way. had to validate a specific job (in KOTS's case it's `validate-success` which is also a branch protection rule). i'd recommend adding one.